cdc/sink/buffer_sink.go

```diff
@@ -151,6 +153,8 @@ func (b *bufferSink) FlushRowChangedEvents(ctx context.Context, resolvedTs uint6
 	case <-ctx.Done():
 		return atomic.LoadUint64(&b.checkpointTs), ctx.Err()
 	case b.flushTsChan <- resolvedTs:
+		return atomic.LoadUint64(&b.checkpointTs), nil
+	default:
 	}
-	return atomic.LoadUint64(&b.checkpointTs), nil
+	return atomic.LoadUint64(&b.checkpointTs), cerror.ErrFlushTsBlocking.FastGenByArgs()
```

Comment:
Is the returned timestamp meaningful when there is a non-nil error?

Reply:
yes, we can ignore the `ErrFlushTsBlocking` in current implementation, which means that the caller will ignore the error and just accept the `checkpointTs` that not updated.

Reply:
When is it necessary to differentiate between "blocking" and "checkpoint-ts not updating"?

Reply:
since n.checkpoints is refreshed asynchronously in a background goroutine,  we return a n.checkpoints every time FlushRowChangedEvents be called, the caller will always check the returned checkpointTs